hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -965,8 +965,9 @@ object HoodieSparkSqlWriter {
     // we must invalidate this table in the cache so writes are reflected in later queries
     if (metaSyncEnabled) {
       getHiveTableNames(hoodieConfig).foreach(name => {
-        val qualifiedTableName = String.join(".", hoodieConfig.getStringOrDefault(HIVE_DATABASE), name)
-        if (spark.catalog.tableExists(qualifiedTableName)) {
+        val syncDb = hoodieConfig.getStringOrDefault(HIVE_DATABASE)
+        val qualifiedTableName = String.join(".", syncDb, name)
+        if (spark.catalog.databaseExists(syncDb) && spark.catalog.tableExists(qualifiedTableName)) {
```

Comment:
`spark.catalog.tableExists(qualifiedTableName)` will contain dbName to check table, why need check db before?

Reply:
Hello, in Spark 3.1 and earlier versions, when detecting the existence of a table, it is mandatory for the database to exist. However, here, the database is not registered in the catalog in advance

Reply:
on, correct me if I'm wrong, it's mean if I use `default` as dbName to check, and it throw error, so check db first is reasonable. But I have a question, in which scenario is this parameter `META_SYNC_DATABASE_NAME` not set, I think we need fix it

Reply:
Okay, I think what you're saying makes sense. Let me see where to complete the registration of databases and tables

Reply:
@danny0405 @KnightChess Hello, the invalidate table operation should only be executed when enableHiveSupport() is enabled, but sometimes we do not need to enable enableHiveSupport(), such as for testing purposes.